modules.d/03systemd-hostnamed/module-setup.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# called by dracut
+check() {
+    [[ $mount_needs ]] && return 1
+
+    if ! dracut_module_included "systemd"; then
+        derror "systemd-hostnamed needs systemd in the initramfs"
+        return 1
+    fi
+
+    return 0
+}
+
+# called by dracut
+depends() {
+    echo "systemd systemd-networkd"
```

Comment:
I don't think this module should depend on systemd-networkd. I did a quick try and it seems hostnamed works fine without networkd.

NetworkManager, when running in the initramfs, currently sets the hostname directly in the kernel. However we are planning to run NM as a systemd service with D-Bus support; in that case it would preferably set the hostname via D-Bus through hostnamed.

In my opinion it should be possible to make other services depend on hostnamed without pulling networkd into the initrd.